src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/prisonersearch/PrisonerSearchResultsProcessor.kt

```diff
@@ -0,0 +1,104 @@
+package uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.prisonersearch
+
+import com.microsoft.applicationinsights.TelemetryClient
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.prisonersearch.model.PagePrisoner
+import uk.gov.justice.digital.hmpps.prisonersearch.model.Prisoner
+import uk.gov.justice.digital.hmpps.prisonersearch.model.PrisonerMatches
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.prisonersearch.PrisonerSearchDataSource.GLOBAL_SEARCH
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.prisonersearch.PrisonerSearchDataSource.MATCH_PRISONERS
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.prisonersearch.PrisonerSearchType.NAME_DOB
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.prisonersearch.PrisonerSearchType.PRISON_NUMBER
+
+@Component
+class PrisonerSearchResultsProcessor(
```

Comment:
New class - a 'processor' that processes the results from the prison offender searches; called by `PrisonSearchService`
At the moment it does 2 things - logs the details of the prisoners using the logger class above ☝️ ; and works out the best match(es) (`PrisonerMatchCount` class) and uses that data to send to App Insights as custom dimensions

Reply:
Once we know more about the data this processor class will likely change into what we really need it to do (or indeed be deleted altogether and allow the service to do what it needs to)